okio/src/commonMain/kotlin/okio/FileSystem.kt

```diff
@@ -164,6 +185,26 @@ abstract class FileSystem {
   @Throws(IOException::class)
   abstract fun sink(file: Path): Sink
 
+  /**
+   * Creates a sink to write [file], executes [writerAction] to write it, and then closes the sink.
+   * This is a compact way to write a file.
+   */
+  inline fun <T> write(file: Path, writerAction: BufferedSink.() -> T): T {
```

Comment:
I use append much less, but it seems like top level given incrementally built output like log files.  Should this be a flag?

Reply:
I claim that most of the time when you want to append, you want to open a Sink, and keep it open for a while, probably in a property of some managing object like `Appender`. The `write` function is most useful when you want to write and immediately close.

Reply:
(We can add an `append` function later if necessary!)